### PR TITLE
Experimental: baking channel to vertex color can support Alpha channel

### PR DESCRIPTION
**Pull request by morirain**

### CODE REVIEW

**Comment by morirain** [comment]
Test: 4.0, 2.93
I did a little work on your code :), It basically creates a temp vertex color, then bakes it twice, and merges the results.

**Comment by morirain** [comment]
Reference #78

**Comment by ucupumar** [comment]
Thanks for the pull request. Ok, I'll test it first 👍

File: bake_common.py [hunk 1]
```diff
@@ -1168,8 +1168,41 @@ def bake_to_vcol(mat, node, root_ch, extra_channel=None, extra_multiplier=1.0):
 
     mat.node_tree.links.new(rgb, emit.inputs[0])
 
+    # To avoid duplicate code, define the function here
+    def bake_alpha_to_vcol():
+        origin_vcol_name = ''
+        for obj in objs:
+            # Gets the origin vertex color of the objs, Their names are normally the same
+            origin_vcol_name = get_active_vertex_color(obj).name
```
Comment (ucupumar) [inline bake_common.py:1176]: I think you should pass bake vcol name to bake_to_vcol function and try to get the vcol based on that name. I think it's safer that way.

**Comment by ucupumar** [comment]
@morirain I just reviewed your code. Can you make a tweak with your code?

**Comment by morirain** [comment]
> @morirain I just reviewed this pull request. Can you make a tweak to your code?

@ucupumar Okay. Any suggestions :)?

**Review by ucupumar — CHANGES_REQUESTED** [review]

**Comment by ucupumar** [comment]
I thought I already pressed the review button. There we go 😁

**Comment by morirain** [comment]
OK, I'll start later, and then I'll test it with a model with a lot of vertices👌

**Comment by morirain** [comment]
The updated code has been uploaded, then i tested two models, one with 2 million vertices in 30 seconds and one with 500,000 vertices in 7 seconds.

**Comment by ucupumar** [comment]
Ok, good enough, I'll merge this pull request, thanks for your contribution! 🙏

**Pull request merged**